### ISSUE TITLE
Problems on Linux (graphic glitches, crashes, writing stops working)

### ISSUE THREAD

**Issue by devingDev** [hzqst/MetaHookSv#518]
Running Fedora 40 x11 Gnome
On a NVidia 3090, Intel 13600KF

Hey,
1.its so noisy, and the ground has those black lines that move when camera moves
![image](https://github.com/user-attachments/assets/c254d5c2-41e2-4eb9-9bee-64c61eb5679c)
![image](https://github.com/user-attachments/assets/d25b58b0-6bad-4212-85c9-ed26830f1915)
![image](https://github.com/user-attachments/assets/1b30076c-1dfe-4e68-82b6-e78d2fac3f2c)

2. Also crashing a lot easily when changing options.
3. Alt+Tab once => can not write in console or in chat (can still move and play otherwise)

Tried different compat, for example: ge proton 9-16 but same results. 
And of course native linux version does not work at all with this (sadly)

Any help or will I have to play without this?

**Comment by hzqst** [comment]
1. same issue as #489 and #464 (turning `r_ssao` off or changing fov to 90 mitigates the problem), and will be refactored soon.

2. then don't change options.

4. that's because the CaptionMod uses a hacky/tricky way to handle IME input events. Disabling CaptionMod.dll in `Half-Life\<mod_directory>\metahook\configs\plugins.lst` mitigates the problem. I will fix this issue soon by using a better approach to handle IME stuffs.

**Comment by devingDev** [comment]
Alright thanks for the quick answer, I'll try those commands then but kind of a bummer that it's either fov or ssao.

2nd is weird but i guess i can set them in mainmenu.
also looks like antialiasing is cut off in the menu, not sure if there is more under it.

3rd i see i will turn it off for now

**Comment by devingDev** [comment]
Ok so 1) default_fov 90   works fine
2nd dunno, it also crashes when i just exit the game lol (shows the crash report sent dialog)

3rd: still not working! i removed the captionmod line (confirmed in options that tab is now gone)
but as soon as i alt tab i can not write anymore. Guess I gotta lock into sven coop until a fix is found then :sob:

**Comment by devingDev** [comment]
alt+tab ""works"" if i remove VGUIExtensions and BulletPhysics(it cries about missing vgui)
but now the options are gone and idk why bulletphysics relies on the vgui kinda weird

**Comment by devingDev** [comment]
Played on a server and sometimes it just crashes out of the blue, no idea what triggers it. I don't get any error or anything. The game just closes and is gone.

**Comment by devingDev** [comment]
Still getting this issue, anything I can look at to find out the culprit? I mean there isn't even an error message. It just closes the game.

**Comment by devingDev** [comment]
Crashed again and first time I got a somewhat "meaningful" error:
![image](https://github.com/user-attachments/assets/ada5238c-24da-4437-963a-8febfbbc8202)

**Comment by hzqst** [comment]
> Crashed again and first time I got a somewhat "meaningful" error: ![image](https://private-user-images.githubusercontent.com/17065236/402372031-ada5238c-24da-4437-963a-8febfbbc8202.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3NTM0MzUsIm5iZiI6MTczNjc1MzEzNSwicGF0aCI6Ii8xNzA2NTIzNi80MDIzNzIwMzEtYWRhNTIzOGMtMjRkYS00NDM3LTk2M2EtOGZlYmZiYmM4MjAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDA3MjUzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1YTQ1NDNmMWYyZTU2OTkzMmVmZjQ0OWFmZmY0ZGM1MzNhNmMxMGU3MGVjMjdmNjY4MzYxODIwZDQ3Mjk4ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.k4QMjDWjoZETPcjRWmhBzbWsbuA5-aB5mm5-sTlSre4)

What's your GPU?

**Comment by KaniEMUACH** [comment]
> Played on a server and sometimes it just crashes out of the blue, no idea what triggers it. I don't get any error or anything. The game just closes and is gone. Happens way too often, not playable at all :(

I have this exact problem though im running Windows 11 with Ryzen 5 5600, 32gb 3600mhz and RX 6600

Removing the line renderer.dll from the plugins.lst file fixed the issue for me

**Comment by devingDev** [comment]
> What's your GPU? I've tested on latest steam-proton with RTX 4060 and RX6800 without problem
rtx 3090


> I have this exact problem though im running Windows 11 with Ryzen 5 5600, 32gb 3600mhz and RX 6600
> 
> Removing the line renderer.dll from the plugins.lst file fixed the issue for me
yea but isn't the renderer one of the major things people use this in svencoop to not have super low fps with certain models etc?

**Comment by KaniEMUACH** [comment]
> Yea but isn't the renderer one of the major things people use this in svencoop to not have super low fps with certain models etc? And interesting that it happens on windows 11 as well.. hmm

Yeah, I think it got something to do with the Server I'm joining in. This server in particular got INS2 weapons modded to the server, so that probably it. But on normal servers, everything works fine.

**Comment by hzqst** [comment]
> > What's your GPU? I've tested on latest steam-proton with RTX 4060 and RX6800 without problem
> 
> RTX 3090, which OS do you use? I'm on Fedora 40 but not sure what could cause this issue anyway.
> 
> > I have this exact problem though im running Windows 11 with Ryzen 5 5600, 32gb 3600mhz and RX 6600
> > Removing the line renderer.dll from the plugins.lst file fixed the issue for me
> 
> Yea but isn't the renderer one of the major things people use this in svencoop to not have super low fps with certain models etc? And interesting that it happens on windows 11 as well.. hmm

I'm on Windows 11 24H2, ubuntu 20.04 LTS and ubuntu 22.04 LTS (ofc as vmware guests)

**Comment by KaniEMUACH** [comment]
> > Yeah, I think it got something to do with the Server I'm joining in. This server in particular got INS2 weapons modded to the server, so that probably it. But on normal servers, everything works fine.
> 
> Same here was on an empty server for like 15 minutes and on another server (called US something) with around 20 people for about 30 minutes no crash then I joined the king/sbmesh ALL maps server and crashed within like 5 minutes, so yeah it seems server related? or maybe some player on there having a specific model/animation or whatever. i'm not sure...

For me i keep crashing while playing or everytime i load the map, the server I play are these two, specifically the 0ap server where the INS2 weapons are.
![image](https://github.com/user-attachments/assets/f7c55733-4f9d-4673-9998-d993a081f899)

I'm gonna do some more testing before imma open a new issue in this repository

**Comment by hzqst** [comment]
"Alt+Tab once => can not write in console or in chat (can still move and play otherwise)"

Should've been fixed in https://github.com/hzqst/MetaHookSv/releases/tag/v20250126e
